### DISCUSSION TITLE
How to define a constant used in multiple expressions?

### DISCUSSION THREAD

**Discussion by An4el** [idaholab/moose#28238] [Q&A Modules: Phase field] [answered]
An general issue is that boudary width is a parameter in initial condition, and it also plays a role in expression defining interphase interfacial parameters (or, kappa in the tutorial). How can I use it in multiple expressions (or, something on the right of "=") so that I don't have to edit a bunch of numbers when I adjusting my model?
I tried GobalParams and GenericConstantMaterial in Materals, but failed to find the right syntax.

for example, my kappa is equal to 3 * sqrt(2) * int_width * other things, how can I achieve it in my input file?

**Comment by GiudGiud** [comment, marked as answer]
Hello

You can do this at the top of the input file
```
kappa = '${fparse 3 * sqrt(2) * int_width * other things}'
```

then in the rest
```
[Some system]
  [some class]
    kappa = ${kappa}
...
```

**Reply by An4el** [reply]
Hmm...Actually my question is how to define 'int_width' in the input file.
Or just explain it in another way, I want to define an global constant x, and there is two parameter y=2x+1 and z=3x in different classes.
I wonder how to define x to ensure y and z correctly found it when defining.

**Reply by An4el** [reply]
Oh I may figure out how it works.
`x = some value`
and just don't put it in any system.
then
```
[Some system]  
  [some class]  
    y = 2 * ${x} + 1
```
It's kind of weird that there could be something put outside of any system.

**Reply by GiudGiud** [reply]
it should be
```
    y = ${fparse 2 * x + 1}
```
the capability is documented. It's very useful
https://mooseframework.inl.gov/application_usage/input_syntax.html
